### PR TITLE
Reduce lambda runtime

### PR DESCRIPTION
**Pull request by mbj**

### CODE REVIEW

File: lambda-runtime/src/AWS/Lambda/Runtime.hs [hunk 1]
```diff
@@ -4,14 +4,15 @@ module AWS.Lambda.Runtime
...
+import AWS.Lambda.Runtime.Types
 import Control.Monad.Except (runExceptT)
 
 import qualified AWS.Lambda.Runtime.Client as Client
 import qualified Data.Aeson                as JSON
 
 run
   :: forall m . (MonadCatch m, MonadIO m)
-  => (JSON.Value -> m JSON.Value)
+  => (LambdaEvent -> m JSON.Value)
```
Comment (epicallan) [inline lambda-runtime/src/AWS/Lambda/Runtime.hs:15]: The `requestId` in `LambdaEvent` can also be retrieved from request headers after parsing `JSON.Value`. Not, that I am opposing exposing `LambdaEvent`

Comment (mbj) [reply lambda-runtime/src/AWS/Lambda/Runtime.hs:15]: you mean response headers. But: at this point its already parsed so why not expose it.

File: lambda-runtime/src/AWS/Lambda/Runtime.hs [hunk 2]
```diff
@@ -13,27 +13,27 @@ run
   :: forall m . (MonadCatch m, MonadIO m)
   => (JSON.Value -> m JSON.Value)
   -> m ()
-run = forever . processEvent
+run function = do
+  config <- either throwM pure =<< liftIO (runExceptT Client.getHttpConfig)
```
Comment (epicallan) [inline lambda-runtime/src/AWS/Lambda/Runtime.hs]: This is a really good fix. Nice. I didn't see it.

Now, that we are re-using the same HTTP manager for potentially more requests. I think we may need to fine-tune the manager to be a bit. Potentially change `managerConnCount  & managerIdleConnectionCount` values.

Comment (mbj) [reply lambda-runtime/src/AWS/Lambda/Runtime.hs]: there can only ever be one event be in flight. So the current values are exactly right.

**Pull request merged**